### PR TITLE
Patrons gamerule

### PR DESCRIPTION
**Pull request by MemeWizard**
- New game rule added:
  - Patron characters will receive a special patron trait
  - Patron characters will not receive a special patron trait

### CODE REVIEW

**Comment by geffe1** [comment]
Isn't it sort of rude to remove such characters, when people actually pay for them to be in the mod? Even as a gamerule, I find that sort of.. discouraging.

File: history/characters/310000_patron_characters.txt [hunk 1]
```diff
@@ -6,7 +6,7 @@
 	culture=lordaeronian religion=holy_light
 	martial=8 diplomacy=4 stewardship=7 intrigue=5 learning=4
 	add_trait=skilled_tactician
-	add_trait=patron
+	add_trait=patron effect={purge_patron_trait_character_effect=yes}
```
Comment (zumbak04) [inline history/characters/310000_patron_characters.txt]: image
Isn't it easier to make `on_startup` event?

File: history/characters/11000_dark_iron.txt [hunk 2]
```diff
@@ -2830,7 +2830,7 @@
 	culture=dark_iron religion=holy_light
 	martial=5 diplomacy=5 stewardship=5 intrigue=5 learning=5
 	trait=elusive_shadow
-	trait=architect trait=sorcerer trait=cynical trait=ambitious trait=class_shaman_7 trait=quick trait=erudite trait=patron
+	trait=architect trait=sorcerer trait=cynical trait=ambitious trait=class_shaman_7 trait=quick trait=erudite trait=add_trait=patron effect={purge_patron_trait_character_effect=yes}
```
Comment (zumbak04) [inline history/characters/11000_dark_iron.txt]:
```suggestion
	trait=architect trait=sorcerer trait=cynical trait=ambitious trait=class_shaman_7 trait=quick trait=erudite add_trait=patron effect={purge_patron_trait_character_effect=yes}
```

File: history/characters/16000_gnome.txt [hunk 3]
```diff
@@ -35,7 +35,7 @@
 	martial=6 diplomacy=6 stewardship=6 intrigue=6 learning=6
 	add_trait=mastermind_theologian
 	add_trait=class_priest_7
-	add_trait=paranoid add_trait=kind add_trait=patron
+	add_trait=paranoid add_trait=kind add_trait=add_trait=patron effect={purge_patron_trait_character_effect=yes}
```
Comment (zumbak04) [inline history/characters/16000_gnome.txt]:
```suggestion
	add_trait=paranoid add_trait=kind add_trait=patron effect={purge_patron_trait_character_effect=yes}
```

File: history/characters/42000_high_elf.txt [hunk 4]
```diff
@@ -1326,7 +1326,7 @@
 	martial=6 diplomacy=6 stewardship=6 intrigue=6 learning=6
 	add_trait=tough_soldier
 	add_trait=class_warrior_7
-	add_trait=quick add_trait=lustful add_trait=ambitious add_trait=slothful add_trait=deceitful add_trait=patron
+	add_trait=quick add_trait=lustful add_trait=ambitious add_trait=slothful add_trait=deceitful add_trait=add_trait=patron effect={purge_patron_trait_character_effect=yes}
```
Comment (zumbak04) [inline history/characters/42000_high_elf.txt]:
```suggestion
	add_trait=quick add_trait=lustful add_trait=ambitious add_trait=slothful add_trait=deceitful add_trait=patron effect={purge_patron_trait_character_effect=yes}
```

File: history/characters/42000_high_elf.txt [hunk 5]
```diff
@@ -1347,7 +1347,7 @@
 	martial=6 diplomacy=6 stewardship=6 intrigue=6 learning=6
 	add_trait=intricate_webweaver
 	add_trait=class_hunter_6
-	add_trait=quick add_trait=envious add_trait=humble add_trait=chaste add_trait=strategist add_trait=patron
+	add_trait=quick add_trait=envious add_trait=humble add_trait=chaste add_trait=strategist add_trait=add_trait=patron effect={purge_patron_trait_character_effect=yes}
```
Comment (zumbak04) [inline history/characters/42000_high_elf.txt]:
```suggestion
	add_trait=quick add_trait=envious add_trait=humble add_trait=chaste add_trait=strategist add_trait=patron effect={purge_patron_trait_character_effect=yes}
```

File: history/characters/42000_high_elf.txt [hunk 6]
```diff
@@ -1361,7 +1361,7 @@
 	martial=6 diplomacy=6 stewardship=6 intrigue=6 learning=6
 	add_trait=martial_cleric
 	add_trait=class_mage_1
-	add_trait=quick add_trait=paranoid add_trait=patient add_trait=humble add_trait=zealous add_trait=patron
+	add_trait=quick add_trait=paranoid add_trait=patient add_trait=humble add_trait=zealous add_trait=add_trait=patron effect={purge_patron_trait_character_effect=yes}
```
Comment (zumbak04) [inline history/characters/42000_high_elf.txt]:
```suggestion
	add_trait=quick add_trait=paranoid add_trait=patient add_trait=humble add_trait=zealous add_trait=patron effect={purge_patron_trait_character_effect=yes}
```

File: history/characters/42000_high_elf.txt [hunk 7]
```diff
@@ -1391,7 +1391,7 @@
 	culture=high_elf religion=cult_of_sunwell
 	martial=5 diplomacy=6 stewardship=4 intrigue=6 learning=5
 	add_trait=midas_touched
-	add_trait=humble add_trait=zealous add_trait=slothful add_trait=patron
+	add_trait=humble add_trait=zealous add_trait=slothful add_trait=add_trait=patron effect={purge_patron_trait_character_effect=yes}
```
Comment (zumbak04) [inline history/characters/42000_high_elf.txt]:
```suggestion
	add_trait=humble add_trait=zealous add_trait=slothful add_trait=patron effect={purge_patron_trait_character_effect=yes}
```

File: history/characters/42000_high_elf.txt [hunk 8]
```diff
@@ -1404,7 +1404,7 @@
 	culture=high_elf religion=cult_of_sunwell
 	martial=5 diplomacy=3 stewardship=7 intrigue=6 learning=6
 	add_trait=elusive_shadow
-	add_trait=patron
+	add_trait=add_trait=patron effect={purge_patron_trait_character_effect=yes}
```
Comment (zumbak04) [inline history/characters/42000_high_elf.txt]:
```suggestion
	add_trait=patron effect={purge_patron_trait_character_effect=yes}
```

File: history/characters/42000_high_elf.txt [hunk 9]
```diff
@@ -1435,7 +1435,7 @@
 	culture=high_elf religion=cult_of_sunwell
 	martial=5 diplomacy=7 stewardship=7 intrigue=7 learning=1
 	add_trait=mastermind_theologian
-	add_trait=diligent add_trait=patient add_trait=shrewd add_trait=patron
+	add_trait=diligent add_trait=patient add_trait=shrewd add_trait=add_trait=patron effect={purge_patron_trait_character_effect=yes}
```
Comment (zumbak04) [inline history/characters/42000_high_elf.txt]:
```suggestion
	add_trait=diligent add_trait=patient add_trait=shrewd add_trait=patron effect={purge_patron_trait_character_effect=yes}
```

File: history/characters/60000_lordaeronian.txt [hunk 10]
```diff
@@ -3701,7 +3701,7 @@
 	add_trait=brilliant_strategist
 	
 	add_trait=class_warrior_7
-	add_trait=wroth add_trait=brave add_trait=ambitious add_trait=shrewd add_trait=greedy add_trait=patron
+	add_trait=wroth add_trait=brave add_trait=ambitious add_trait=shrewd add_trait=greedy add_trait=add_trait=patron effect={purge_patron_trait_character_effect=yes}
```
Comment (zumbak04) [inline history/characters/60000_lordaeronian.txt]:
```suggestion
	add_trait=wroth add_trait=brave add_trait=ambitious add_trait=shrewd add_trait=greedy add_trait=patron effect={purge_patron_trait_character_effect=yes}
```

File: history/characters/60000_lordaeronian.txt [hunk 11]
```diff
@@ -3910,7 +3910,7 @@
 	add_trait=scholarly_theologian
 	
 	add_trait=class_priest_6
-	add_trait=shrewd add_trait=patient add_trait=humble add_trait=ambitious add_trait=cynical add_trait=patron
+	add_trait=shrewd add_trait=patient add_trait=humble add_trait=ambitious add_trait=cynical add_trait=add_trait=patron effect={purge_patron_trait_character_effect=yes}
```
Comment (zumbak04) [inline history/characters/60000_lordaeronian.txt]:
```suggestion
	add_trait=shrewd add_trait=patient add_trait=humble add_trait=ambitious add_trait=cynical add_trait=patron effect={purge_patron_trait_character_effect=yes}
```

File: history/characters/310000_patron_characters.txt (hunk 1 above)
Comment (MemeWizard) [reply history/characters/310000_patron_characters.txt]: No

File: common/game_rules/wc_unimportant_game_rules.txt [hunk 12]
```diff
@@ -1,3 +1,17 @@
...
+	option = {
+		name = exist
+		text = "RULE_PATRONS_EXIST"
+		desc = "RULE_PATRONS_EXIST_DESC"
+	}
+	option = {
+		name = patron_trait
+		text = "RULE_PATRONS_PATRON_TRAIT"
+		desc = "RULE_PATRONS_PATRON_TRAIT_DESC"
+	}
```
Comment (zumbak04) [inline common/game_rules/wc_unimportant_game_rules.txt]: Strange that `patron_trait` isn't default rule.

**Comment by geffe1** [comment]
Works and would approve, but I do not like the wording here.

https://gyazo.com/2a792a4311189b43e70fba9e55e52e89
image

Why not just make it "Patron characters receive a special patron trait"
On/Off

**Comment by MemeWizard** [comment]
Changed localization

File: common/game_rules/wc_unimportant_game_rules.txt (hunk 12 above)
Comment (MemeWizard) [reply common/game_rules/wc_unimportant_game_rules.txt]: Changed, but, imo, "by default" this trait feels like it's off-place and breaks immersion

**Pull request merged**